src/unit/test_rax.c

```diff
@@ -1023,3 +1023,55 @@ int test_raxFuzz(int argc, char **argv, int flags) {
     }
     return !!errors;
 }
+
+/* Test for crash in raxRemove when two keys share a common prefix in the same rax
+ * structure. Upon deletion of one key, rax attempts to recompress the structure
+ * back to its original form for other key. The crash occurs when the
+ * recompression exceeds 512MB size limit due to a missing size check in the
+ * compression logic.
+ *
+ * This test is disabled by default because it uses a lot of memory. */
+int test_raxRemoveCrash(int argc, char **argv, int flags) {
```

Comment:
"Remove crash" is not a very descriptive title. :D

The comment above says things like "The crash occurs" but it should instead describe that the test checks that a crash doesn't occur. It can say that in the past, there was a crash.

```suggestion
int test_raxRecompressHugeKey(int argc, char **argv, int flags) {
```

Reply:
Thank you for the quick response and the review. I have addressed your suggestions.